### PR TITLE
encrypt-spring-boot-jpa-example

### LINKED ISSUE #15810: [OpenForce 2022]  Add `encrypt-spring-boot-jpa-example` for `encrypt-example`
Hi community,

This issue is for [OpenForce 2022](https://github.com/apache/shardingsphere/discussions/12683#discussioncomment-2139441)

### Background

[encrypt-example](https://github.com/apache/shardingsphere/tree/master/examples/shardingsphere-jdbc-example/single-feature-example/encrypt-example) is used to demonstrate how to integrate ShardingSphere-JDBC and encrypt data.

Now, it has three sub-modules:

```
<module>encrypt-raw-jdbc-example</module>
<module>encrypt-spring-boot-mybatis-example</module>
<module>encrypt-spring-namespace-mybatis-example</module>
```
### Aim

Add `encrypt-spring-boot-jpa-example` for `encrypt-example`

### Basic Qualifications

- Java
- Maven

### Example FYI

- [sharding-spring-boot-jpa-example](https://github.com/apache/shardingsphere/tree/master/examples/shardingsphere-jdbc-example/single-feature-example/sharding-example/sharding-spring-boot-jpa-example)

### Notice
- You need to clone the ShardingSphere project from [here](https://github.com/apache/shardingsphere) first
- In the ShardingSphere directory, `examples` is a separate maven project, you should open `examples` instead of `ShardingSphere` in the IDE.
- Contributor Guide: https://shardingsphere.apache.org/community/en/contribute/contributor/
- Code of Conduct: https://shardingsphere.apache.org/community/en/contribute/code-conduct/

### PR DESCRIPTION
**Pull request by Swastyy**
Fixes #15810 

Changes proposed in this pull request:
-
-
-

### CODE REVIEW

**Comment by RaigorJiang** [comment]
Hi @Swastyy 
CI failed,  please fix it.  😀

**Comment by Swastyy** [comment]
> Hi @Swastyy CI failed, please fix it. 😀

Yes @RaigorJiang , I'll complete it. Thanks.

File: examples/shardingsphere-jdbc-example/single-feature-example/encrypt-example/encrypt-spring-boot-jpa-example/src/main/java/org/apache/shardingsphere/example/encrypt/spring/boot/jpa/EncryptSpringBootJpaExample.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.context.annotation.ComponentScan;
+
+import java.sql.SQLException;
+
+public class EncryptSpringBootJpaExample {
+    
+    public static void main(final String[] args) throws SQLException {
+        try (ConfigurableApplicationContext applicationContext = SpringApplication.run(EncryptSpringBootJpaExample.class, args)) {
+            ExampleExecuteTemplate.run(applicationContext.getBean("encrypt", ExampleService.class));
```
Comment (yx9o) [inline examples/shardingsphere-jdbc-example/single-feature-example/encrypt-example/encrypt-spring-boot-jpa-example/src/main/java/org/apache/shardingsphere/example/encrypt/spring/boot/jpa/EncryptSpringBootJpaExample.java:33]: Can I get a bean named `encrypt` here.

File: examples/shardingsphere-jdbc-example/single-feature-example/encrypt-example/encrypt-spring-boot-jpa-example/src/main/java/org/apache/shardingsphere/example/encrypt/spring/boot/jpa/EncryptSpringBootJpaExample.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import org.apache.shardingsphere.example.core.api.ExampleExecuteTemplate;
+import org.apache.shardingsphere.example.core.api.service.ExampleService;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.context.annotation.ComponentScan;
+
+import java.sql.SQLException;
+
+public class EncryptSpringBootJpaExample {
```
Comment (RaigorJiang) [inline examples/shardingsphere-jdbc-example/single-feature-example/encrypt-example/encrypt-spring-boot-jpa-example/src/main/java/org/apache/shardingsphere/example/encrypt/spring/boot/jpa/EncryptSpringBootJpaExample.java:29]: Hi @Swastyy , I think this class needs some annotations, please refer to `ReadwriteSplittingSpringBootJpaExample` and `ShardingSpringBootJpaExample`.

**Review by RaigorJiang — CHANGES_REQUESTED** [review]
Need a properties file for spring in resources, common name is `application.properties`.
Please refer to [readwrite-splitting-example](https://github.com/apache/shardingsphere/tree/master/examples/shardingsphere-jdbc-example/single-feature-example/readwrite-splitting-example/readwrite-splitting-spring-boot-jpa-example/src/main/resources)

**Comment by Swastyy** [comment]
Ok, I will update @RaigorJiang and @yx9o .

**Comment by RaigorJiang** [comment]
Hi @Swastyy 
Sorry, there are some file conflicts, please fix them.

**Comment by Swastyy** [comment]
> Hi @Swastyy Sorry, there are some file conflicts, please fix them.

Hi @RaigorJiang , the option for resolve conflicts is not working

**Comment by RaigorJiang** [comment]
Sorry, there are a lot of affected files (some test cases) that don't seem to be related to example, can you update the PR after merging master locally?

**Comment by Swastyy** [comment]
> Sorry, there are a lot of affected files (some test cases) that don't seem to be related to example, can you update the PR after merging master locally?

Ok @RaigorJiang , I will do this.

**Comment by Swastyy** [comment]
Hi @RaigorJiang , I am closing this PR, and will make a new PR, it will be also easy to review then.

**Comment by RaigorJiang** [comment]
@Swastyy 
OK, thank you!

**Pull request closed without merging**